core/error_macros.h

```diff
@@ -118,411 +99,550 @@ void _err_print_index_error(const char *p_function, const char *p_file, int p_li
 			return m_retval;                                                                                                         \
 		}                                                                                                                            \
-	}
+	} while (0)
 
 /**
- * If `m_cond` evaluates to `true`, prints a custom error message and returns the value specified in `m_retval`.
+ * Ensures `m_cond` is false.
+ * If `m_cond` is true, prints `m_msg` and the current function returns `m_retval`.
+ *
+ * If checking for null use ERR_FAIL_NULL_V_MSG instead.
+ * If checking index bounds use ERR_FAIL_INDEX_V_MSG instead.
  */
 #define ERR_FAIL_COND_V_MSG(m_cond, m_retval, m_msg)                                                                                                   \
-	{                                                                                                                                                  \
+	do {                                                                                                                                               \
 		if (unlikely(m_cond)) {                                                                                                                        \
-			_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Condition \"" _STR(m_cond) "\" is true. Returned: " _STR(m_retval), DEBUG_STR(m_msg)); \
+			_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Condition \"" _STR(m_cond) "\" is true. returned: " _STR(m_retval), DEBUG_STR(m_msg)); \
 			return m_retval;                                                                                                                           \
 		}                                                                                                                                              \
-	}
+	} while (0)
 
 /**
- * If `m_cond` evaluates to `true`, prints a custom error message and continues the loop the macro is located in.
+ * Try using `ERR_CONTINUE_MSG`.
+ * Only use this macro if there is no sensible error message.
+ *
+ * Ensures `m_cond` is false.
+ * If `m_cond` is true, the current loop continues.
  */
 #define ERR_CONTINUE(m_cond)                                                                                           \
-	{                                                                                                                  \
+	do {                                                                                                               \
 		if (unlikely(m_cond)) {                                                                                        \
 			_err_print_error(FUNCTION_STR, __FILE__, __LINE__, "Condition \"" _STR(m_cond) "\" is true. Continuing."); \
 			continue;                                                                                                  \
 		}                                                                                                              \
-	}
+	} while (0)
```

Comment:
`continue` will end inner loop, but not the outer one (same below in `ERR_CONTINUE_MSG`, some in breaks `ERR_BREAK` and `ERR_BREAK_MSG`).

Reply:
Good point. I'll submit a PR to wrap these in `if(0) { } else` instead.

Reply:
> EDIT: Actually it's easier to just add a dangling else to the existing if (unlikely(m_cond))

But code such as this would still be valid then?
```c++
ERR_CONTINUE(x != 0) // No semicolon
z()
```

Maybe add a trailing `do {} while(0)`?

Reply:
> But code such as this would still be valid then?

Actually the compiler warns about the dangling else; so it probably needs to be `if(unlikely(m_cond)) { } else ((void)0)`